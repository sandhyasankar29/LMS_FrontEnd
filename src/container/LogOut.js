import { Logout } from 'grommet-icons';
import { React } from 'react';
import { Button } from 'grommet';
import { useHistory } from "react-router-dom";
const LogOut = (props) => {
    const history = useHistory();
    return (<Button pad='xsmall' id="logoutBtn" plain icon={<Logout color="red" />} onClick={() => { localStorage.clear(); history.replace(props.route) }} ><Logout /></Button>)
}

export default LogOut;

