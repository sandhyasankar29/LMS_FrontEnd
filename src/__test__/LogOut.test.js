import React from 'react';
import ReactDOM from 'react-dom';
import LogOut from '../container/LogOut';
import { cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'

afterEach(cleanup)
it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<LogOut></LogOut>, div)
})

it("matches snapshot", () => {
    const tree = renderer.create(<LogOut></LogOut>).toJSON();
    expect(tree).toMatchSnapshot();
})
