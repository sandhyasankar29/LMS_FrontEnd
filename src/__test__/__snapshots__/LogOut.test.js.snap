// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<button
  className="StyledButton-sc-323bzc-0 bZMlsb"
  id="logoutBtn"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  type="button"
>
  <svg
    aria-label="Logout"
    className="StyledIcon-ofa7kd-0 eBBIEF"
    viewBox="0 0 24 24"
  >
    <path
      d="M13,9 L13,2 L1,2 L1,22 L13,22 L13,15 M22,12 L5,12 M17,7 L22,12 L17,17"
      fill="none"
      stroke="#231F20"
      strokeWidth="2"
    />
  </svg>
</button>
`;
